src/main/kotlin/org/cascadebot/cascadebot/messaging/Messaging.kt

```diff
@@ -113,6 +114,68 @@ object Messaging {
+        val pageGroup = GuildDataManager.getGuildData(textChannel.guild.idLong).pageCache[message.idLong]
+        handlePage(textChannel, message, 1, pageGroup!!)
+    })
+
+    private val prevPageButton = UnicodeButton(UnicodeConstants.BACKWARD_ARROW, IButtonRunnable { _: Member?, textChannel: TextChannel, message: Message ->
+        val pageGroup = GuildDataManager.getGuildData(textChannel.guild.idLong).pageCache[message.idLong]
+        val newPage = pageGroup!!.currentPage - 1
+        handlePage(textChannel, message, newPage, pageGroup)
+    })
+
+    private val nextPageButton = UnicodeButton(UnicodeConstants.FORWARD_ARROW, IButtonRunnable { _: Member?, textChannel: TextChannel, message: Message ->
+        val pageGroup = GuildDataManager.getGuildData(textChannel.guild.idLong).pageCache[message.idLong]
+        val newPage = pageGroup!!.currentPage + 1
+        handlePage(textChannel, message, newPage, pageGroup)
+    })
+
+    private val lastPageButton = UnicodeButton(UnicodeConstants.FAST_FORWARD, IButtonRunnable { _: Member?, textChannel: TextChannel, message: Message ->
+        val pageGroup = GuildDataManager.getGuildData(textChannel.guild.idLong).pageCache[message.idLong]
+        handlePage(textChannel, message, pageGroup!!.pages, pageGroup)
+    })
+
+    private fun handlePage(textChannel: TextChannel, message: Message, newPage: Int, pageGroup: PageCache.Pages) {
+        if (newPage < 1 || newPage > pageGroup.pages) {
+            return
+        }
+        pageGroup.getPage(newPage).pageShow(message, newPage, pageGroup.pages)
+        pageGroup.currentPage = newPage
+
+        val group = GuildDataManager.getGuildData(textChannel.guild.idLong).buttonsCache[textChannel.idLong]?.get(message.idLong)
+
+        if (newPage == 1) {
+            group?.removeButton(firstPageButton)
+            group?.removeButton(prevPageButton)
+            if (pageGroup.isMissingButtons) {
+                group?.addButton(nextPageButton)
+            }
+            pageGroup.isMissingButtons = true
+        } else if (newPage == pageGroup.pages) {
+            group?.removeButton(lastPageButton)
+            group?.removeButton(nextPageButton)
+            pageGroup.isMissingButtons = true
+            if (pageGroup.isMissingButtons) {
+                group?.addButton(prevPageButton)
+            }
+        } else if (pageGroup.isMissingButtons) {
+            group?.removeButton(firstPageButton)
+            group?.removeButton(prevPageButton)
+            group?.removeButton(nextPageButton)
+            group?.removeButton(lastPageButton)
+            if (pageGroup.pages > 2) {
+                group?.addButton(firstPageButton)
+            }
+            group?.addButton(prevPageButton)
+            group?.addButton(nextPageButton)
+            if (pageGroup.pages > 2) {
+                group?.addButton(lastPageButton)
+            }
+            pageGroup.isMissingButtons = false
```

Comment:
I'm unsure about this. For two reasons: 
 - The usage of it is fairly clunky when going from page 1 -> 2 and last page -> last page - 1
 - This is a *lot* of rest actions to be doing. I'm not sure it's worth the tiny (imo) UI improvement

Reply:
jvs told me he wanted pages to not have buttons when they shouldn't need them and that i could only do paged usage if i implemented those changes. And I'm fairly sure he'd complain if we didn't reset the buttons. I'll get him input on this though.
@mrjvs Whats your thoughts on this?